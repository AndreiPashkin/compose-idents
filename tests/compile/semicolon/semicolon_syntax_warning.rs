//! Deprecated usage of semicolons as alias definition separators should result in a compile-time
//! deprecation warning.
#![deny(warnings)]
use compose_idents::compose_idents;

compose_idents!(my_fn = concat(foo, _, bar); {
    fn my_fn() -> u32 {
        1
    }
});

compose_idents!(
    my_fn_1 = concat(foo, _, baz);
    my_fn_2 = concat(spam, _, eggs); {
    fn my_fn_1() -> u32 {
        2
    }

    fn my_fn_2() -> u32 {
        3
    }
});

fn main() {
    assert_eq!(foo_bar(), 1);
    assert_eq!(foo_baz(), 2);
    assert_eq!(spam_eggs(), 3);
}
