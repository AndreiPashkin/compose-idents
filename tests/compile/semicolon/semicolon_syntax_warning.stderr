error: use of deprecated function `foo_bar`: compose_idents!: Using semicolons as separators is deprecated, use commas instead
  --> tests/compile/semicolon/semicolon_syntax_warning.rs:25:16
   |
25 |     assert_eq!(foo_bar(), 1);
   |                ^^^^^^^
   |
note: the lint level is defined here
  --> tests/compile/semicolon/semicolon_syntax_warning.rs:3:9
   |
3  | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(deprecated)]` implied by `#[deny(warnings)]`

error: use of deprecated function `foo_baz`: compose_idents!: Using semicolons as separators is deprecated, use commas instead
  --> tests/compile/semicolon/semicolon_syntax_warning.rs:26:16
   |
26 |     assert_eq!(foo_baz(), 2);
   |                ^^^^^^^
